### PR TITLE
cost subsidy implementation

### PR DESCRIPTION
**Pull request by zishanbilal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808)

### CODE REVIEW

**Comment by zishanbilal** [comment]
@colinsheppard @sfwatergit please review, i have incorporated the feedback.

**Comment by zishanbilal** [comment]
@colinsheppard @sfwatergit please review, i have incorporated the feedback.

File: src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala [hunk 1]
```diff
@@ -110,9 +112,13 @@ class ModeChoiceMultinomialLogit(val beamServices: BeamServices, val model: Mult
         case _ =>
           altAndIdx._1.costEstimate
       }
+      val subsidy = modeSubsidy.getSubsidy(mode, age, income)
       val subsidisedCost =
-        Math.max(0, totalCost.toDouble - modeSubsidy.getSubsidy(altAndIdx._1.tripClassifier, age, income))
-      val numTransfers = altAndIdx._1.tripClassifier match {
+        Math.max(0, totalCost.toDouble - subsidy)
+
+      if(totalCost < subsidy) logger.warn("Mode subsidy is even higher then the cost. Mode: {}, Cost: {}, Subsidy: {}", mode, totalCost, subsidy)
```
Comment (colinsheppard) [inline src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala]: This warning should make it clear that zero is assumed and not a negative cost.

**Comment by sfwatergit** [comment]
___
*[src/main/scala/beam/agentsim/agents/choice/mode/ModeSubsidy.scala, line 34 at r3](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808#-LPRuIJ840enO3sb6bIU:-LPTKfv06V_IJq_kJmtv:b6e1jfz) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/68a064bc1948c0371c0c85fdb34283ecbd716f56/src/main/scala/beam/agentsim/agents/choice/mode/ModeSubsidy.scala#L34)):*
<details><summary><i>Previously, colinsheppard (Colin Sheppard) wrote…</i></summary><blockquote>

I like the use of mathematical range notation myself (i.e. they can use [10,20) to indicate 10 inclusive to 20 exclusive)
</blockquote></details>

Ok. Keep the range notation then. Please rename the file to ModeSubsidies.csv

**Comment by sfwatergit** [comment]
___
*[src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceDriveIfAvailable.scala, line 16 at r1](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808#-LPCp5Qt99I9uCqzzmkZ:-LPTM3cWDHLfh0yAV7Vv:bg7igkv) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/49ecc74d5ce7478e24d4d84eebe7512542359af2/src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceDriveIfAvailable.scala#L16)):*
<details><summary><i>Previously, colinsheppard (Colin Sheppard) wrote…</i></summary><blockquote>

done
</blockquote></details>

:lgtm:

**Comment by sfwatergit** [comment]
___
*[test/input/beamville/subsidies.csv, line 0 at r4](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808#-LPTT3B04tQ2jfVdhE2H:-LPTT3B04tQ2jfVdhE2I:b-yqksph) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/4afa9a6c63dbd030a7b3cffcafc931a52da740ae/test/input/beamville/subsidies.csv#L0)):*
You have a typo in your subsidies file. Line 3 should be 
"transit,(50:75],[0,20000],2"

You have;
transit,(50,75],[0,20000],2

**Comment by sfwatergit** [comment]
___
*[test/input/beamville/subsidies.csv, line  at r4](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808#-LPTT3B04tQ2jfVdhE2H:-LPTTZ6f-ad3m3Zwvjnj:bl2sf7r) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/4afa9a6c63dbd030a7b3cffcafc931a52da740ae/test/input/beamville/subsidies.csv#L)):*
> "transit,(50:75],[0,20000],2"
I mean, "transit,(50:75],[0:20000],2"

**Comment by sfwatergit** [comment]
___
*[src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala, line 91 at r1](https://reviewable.io/reviews/lbnl-ucb-sti/beam/808#-LPChV5f9KXtaRDmBgkJ:-LPThHql7dnB8TeT_RjS:bg7igkv) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/49ecc74d5ce7478e24d4d84eebe7512542359af2/src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala#L91)):*
<details><summary><i>Previously, zishanbilal (Zeeshan Bilal) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

:lgtm:

**Pull request merged**